library/common/jni_support.h

```diff
@@ -0,0 +1,7 @@
+#include <jni.h>
+
+// NOLINT(namespace-envoy)
+
+jint platform_setup(JavaVM* jvm, jobject connectivity_manager);
```

Comment:
I would put a comment here for now saying that this initializes ARES on Android and is not called from pure Java (but would be a no-op if it was). Basically, explain this whole thing.